go/flatpak-go-deps.py

```diff
@@ -0,0 +1,282 @@
+            print(f"✨ Found commit ID: {commit_id}")
+            return commit_id
+        except subprocess.CalledProcessError:
+            return None
+
+
+def get_module_info(module_name):
+    result = subprocess.run(
+        ["go", "list", "-m", "-json", module_name],
+        check=True,
+        capture_output=True,
+        text=True,
+    )
+    return json.loads(result.stdout)
+
+
+def get_git_url(module_name):
+    # Remove the version suffix, if present
+    module_name = re.sub(r"/v\d+$", "", module_name)
+
+    # Remove the subdirectory, if present (e.g. github.com/foo/bar/subdir -> github.com/foo/bar)
+    if "gitlab.com" in module_name or "github.com" in module_name:
+        url_parts = module_name.split("/")
+        if len(url_parts) > 3:
+            module_name = "/".join(url_parts[:3])
+
+    if "gitlab.com" in module_name:
+        return f"https://gitlab.com/{module_name.replace('gitlab.com/', '')}"
+    elif "github.com" in module_name:
+        return f"https://github.com/{module_name.replace('github.com/', '')}"
+    elif "git.torproject.org" in module_name:
+        return f"https://{module_name}"
+    else:
+        response = requests.get(f"https://{module_name}/?go-get=1")
+        if response.status_code != 200:
+            return None
+        soup = BeautifulSoup(response.content, "html.parser")
+        meta_tag = soup.find("meta", {"name": "go-import"})
+        if meta_tag:
+            url = meta_tag["content"].split()[2]
+            r = requests.get(url, allow_redirects=True)
+            if r.history:
+                return r.url
+            else:
+                return url
+
+        return None
+
+
+@click.command()
+@click.argument("repo_and_folder", type=str)
+@click.option("--version", default=None, help="Version of the repository.")
+@click.option("--github_api_token", default=None, help="GitHub API Token.")
+@click.option("--gitlab_api_token", default=None, help="GitLab API Token.")
+def main(repo_and_folder, version, github_api_token, gitlab_api_token):
+    """Flatpak Go Generator"""
+
+    if "/" in repo_and_folder:
+        repo, folder = repo_and_folder.rsplit("/", 1)
+        repo_name = folder
+    else:
+        repo = repo_and_folder
+        folder = ""
+        repo_name = os.path.basename(repo).replace(".git", "")
+
+    with tempfile.TemporaryDirectory() as temp_dir:
+        os.chdir(temp_dir)
+
+        print("✨ Creating temporary Go module")
+        subprocess.run(["go", "mod", "init", "tempmod"], check=True)
+
+        try:
+            print("✨ Cloning the target repository")
+            subprocess.run(
+                ["git", "clone", f"https://{repo}", f"src/{repo_name}"], check=True
+            )
+            os.chdir(f"src/{repo_name}")
+
+            if version:
+                print(f"✨ Checking out version {version}")
+                subprocess.run(["git", "checkout", version], check=True)
+
+            os.chdir(temp_dir)
+
+            if folder:
```

Comment:
I think this `if` is a mistake. When given a root dir such as `github.com/a/b.git/`, this script detects no dependencies because it never switches into the repo directory. When I comment this line, it works.

Also it'd be nice if we could just give it the git url without the slash